### PR TITLE
Approval test and pewpew.py thread refactor

### PR DESCRIPTION
**Pull request by objarni**
While adding an approval test (test_pewpew.py) I refactored the "run" function of the thread.

It's been running some hours on my machine and seems to work fine.

@ultrabug you might want to have a look, also at the test_pewpew module. It's a little dirty ATM I think it can be cleaned up quite a bit by pytest decorators (I'm new to pytest so @obestwalter probably pukes a little when he sees the test haha)

### CODE REVIEW

File: tests/test_pewpew.py [hunk 1]
```diff
@@ -1,9 +1,24 @@
...
+    MINOR,
+)
+
+event = namedtuple("event", ["type", "code", "value"])
 
 log = logging.getLogger(__name__)
+logging.basicConfig(level=logging.DEBUG)
 
 
 def test_sanity():
```
Comment (obestwalter) [inline tests/test_pewpew.py]: We can get rid of this one I guess (test_sanity is not really doing anything sensible).

**Review by obestwalter — APPROVED** [review]

**Comment by obestwalter** [comment]
Nice one @objarni.

File: tests/test_pewpew.py (hunk 1 above)
Comment (objarni) [reply tests/test_pewpew.py]: Good spot, done

**Comment by ultrabug** [comment]
I'll test this ASAP unless you already did @obestwalter ?

Shall we move on?

**Comment by obestwalter** [comment]
Didn't test anything but @objarni did :)

**Pull request merged**